### PR TITLE
chore: add extensions for `suggest` in composer.json

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add extentions for `suggest`

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
`ext-xml` is not listed in `composer.json`, but it is checked in `CodeIgniter.php`
I'm not sure `xml` is really needed.

CodeIgniter4/CodeIgniter.php at develop · codeigniter4/CodeIgniter4
https://github.com/codeigniter4/CodeIgniter4/blob/a1a635814eb5c254b5a31cbfde054ff54f0b30a3/system/CodeIgniter.php#L214-L220

**Review by samsonasik — APPROVED** [review]

**Review by MGatner — CHANGES_REQUESTED** [review]
Could you also update **user_guide/intro/requirements.html**? We have some inconsistencies between here and there about what is "required" versus "suggested".

**Comment by MGatner** [comment]
> I'm not sure xml is really needed.

Sorry, missed this comment before review.

**Comment by kenjis** [comment]
> what is "required" versus "suggested".

I think `curl` is suggested. It is needed only if you use CURLRequest.

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,15 @@
...
-        "ext-fileinfo": "Improves mime type detection for files"
+        "ext-imagick": "If you use ImageMagickHandler",
+        "ext-simplexml": "If you format XML",
+        "ext-mysqli": "If you use MySQL",
+        "ext-oci8": "If you use Oracle Database",
+        "ext-pgsql": "If you use PostgreSQL",
+        "ext-sqlsrv": "If you use SQL Server",
+        "ext-sqlite3": "If you use SQLite3",
+        "ext-fileinfo": "Improves mime type detection for files",
+        "ext-readline": "Improves CLI::input() usability"
```
Comment (MGatner) [inline composer.json:38]: I think Composer Normalize will alphabetize these:
```suggestion
        "ext-fileinfo": "Improves mime type detection for files",
        "ext-imagick": "If you use ImageMagickHandler",
        "ext-mysqli": "If you use MySQL",
        "ext-oci8": "If you use Oracle Database",
        "ext-pgsql": "If you use PostgreSQL",
        "ext-readline": "Improves CLI::input() usability",
        "ext-simplexml": "If you format XML",
        "ext-sqlite3": "If you use SQLite3",
        "ext-sqlsrv": "If you use SQL Server"
```

**Review by MGatner — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (kenjis) [reply composer.json:38]: What is *Composer Normalize* ?

Comment (paulbalandan) [reply composer.json:38]: It's a Composer plugin. `ergebnis/composer-normalize`

Comment (kenjis) [reply composer.json:38]: It seems we don't use it.

```console
$ composer normalize
                                       
  Command "normalize" is not defined.  
```

Comment (MGatner) [reply composer.json]: Oops! My mistake. I use it locally and it is part of DevKit so I assumed we had it here, but maybe not. We can always come back and normalize later - merging this now so it makes it into `4.2.2`.

**Pull request merged**